src/main/java/net/neoforged/neoforge/event/village/VillagerChangeProfessionEvent.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.village;
+
+import net.minecraft.world.entity.npc.Villager;
+import net.minecraft.world.entity.npc.VillagerProfession;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.neoforge.common.NeoForge;
+import net.neoforged.neoforge.event.EventHooks;
+import net.neoforged.neoforge.event.entity.EntityEvent;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * VillagerChangeProfessionEvent is fired on the server when a Villager changes profession.
+ * This event is fired via the {@link EventHooks#onVillagerProfessionChanged(Villager, VillagerProfession, VillagerProfession)}.
+ * This event is {@link ICancellableEvent cancellable}. Effects of the cancellation are noted in {@link #setCanceled(boolean)}.
+ * You can change the profession that the villager will change to via {@link #setNewProfession(VillagerProfession)}.
+ * This event is fired on the {@link NeoForge#EVENT_BUS}.
```

Comment:
Please add a blank line between each paragraph, and prepend `<p>` to each paragraph for proper separation.